### DISCUSSION TITLE
Build Error: ./build_files.sh: line 3: python3.10: command not found

### DISCUSSION THREAD

**Discussion by laraib07** [vercel/community#4481] [Help] [answered]
### Summary

Vercel deployment is failing due to `./build_files.sh: line 3: python3.10: command not found`. 
Here is my build file:
```bash
# Build the project
echo "Building the project..."
python3.10 -m pip install -r requirements.txt

echo "Make Migration..."
python3.10 manage.py makemigrations --noinput
python3.10 manage.py migrate --noinput

echo "Collect Static..."
python3.10 manage.py collectstatic --noinput --clear
```

And here is my `vercel.json` file:
```js

{
  "version": 2,
  "builds": [
    {
      "src": "dcrm/wsgi.py",
      "use": "@vercel/python",
      "config": {"maxLambdaSize": "15mb", "runtime": "python3.10"}
    },
    {
      "src": "build_files.sh",
      "use": "@vercel/static-build",
      "config": {
        "distDir": "staticfiles"
      }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "dcrm/wsgi.py"
    },
    {
      "src": "/static/(.*)",
      "dest": "/static/$1"
    }
  ]
}
```

### Steps to Reproduce

1. Clone my repo from [here](https://github.com/laraib07/DCRM/tree/vercel-fix-2)
2. Deploy on vercel (add appropriate Environment Variables)
3. Find out.

**Comment by amyegan** [comment, marked as answer]
The [Python Runtime](https://vercel.com/docs/functions/serverless-functions/runtimes/python#python-version) uses Python `3.9` by default. Support for `3.10` is not yet available. 🙇

**Reply by laraib07** [reply]
I thought it did as the mentioned `You can specify the version of Python to use by defining python_version in Pipfile`.
Will change my `vercel.json` and `build_files.sh` and try again.
Thanks you so much!

**Reply by amyegan** [reply]
Hey. 👋 Just wanted to let you know that we now support Python 3.12

https://vercel.com/docs/deployments/build-image#runtime-support
